### PR TITLE
config: Enable jailer by default when using firecracker

### PR DESCRIPTION
**Pull request by jrieck1991**
Now that we have enabled CI tests for jailed firecracker and we have
fixed the  issue with removing the block storage device #2387, we
should leverage the full power of firecracker and enable jailer by
default.

Fixes: #2455

Signed-off-by: Jack Rieck <jack.rieck@sendgrid.com>

### CODE REVIEW

**Comment by devimc** [comment]
thanks @jrieck1991 - please fix your commit message, take a look to the following document https://github.com/kata-containers/community/blob/main/CONTRIBUTING.md#patch-format

regarding you patch .... I feel that it can break the CI

cc @GabyCT

**Comment by fidencio** [comment]
> regarding you patch .... I feel that it can break the CI
> 
> cc @GabyCT

Howdy! :-)

I'm not sure if the patch will actually break the firecracker CI, as it should just work with the jailer enabled and that's pretty much what the firecracker-with-jailer CI does.

If we turn this switch we should:
* Consider removing the CI running **without** jailer
* Have a huge release note saying this is the new default for firecracker.

Now, let me go through the code and review the patch itself.

File: src/runtime/cli/config/configuration-fc.toml.in
Comment (fidencio) [inline src/runtime/cli/config/configuration-fc.toml.in:35]: This change looks good, but shouldn't we also remove the comments above the line you changed?
"""
# This is disabled by default as additional setup is required
# for this feature today.
"""
^^^^^ Those :-)

**Comment by GabyCT** [comment]
so is this PR going to break the CI? and why we are going to merge this if the CI gets broken?

**Comment by fidencio** [comment]
> so is this PR going to break the CI? and why we are going to merge this if the CI gets broken?

This PR **may** break the CI, but we're not yet discussing whether it'll be merged or not before running the CI.
I sincerely don't think it'll break the CI and if it does, the breakage will be controlled (as in, we well need a patch on the tests side to unbreak it).

Let's run the CI and confirm whether it'll break something or not.  But, for sure, it'll make the firecracker CIs redundant.

**Comment by fidencio** [comment]
So, CI seems to go well with the changes, I specially checked:
* http://jenkins.katacontainers.io/job/kata-containers-2.0-firecracker-ubuntu-PR/988/
* http://jenkins.katacontainers.io/job/kata-containers-2.0-firecracker-jailer-ubuntu-PR/15/

Now, let's go to fixing the commits and the commit message.  I'd recommend:
1. Squash both commts.
2. Make your commit message look like:
```
config: Enable jailer by default when using firecracker

Now that we have enabled CI tests for jailed firecracker and we have
fixed the  issue with removing the block storage device #2387, we
should leverage the full power of firecracker and enable jailer by
default.

Fixes: #2455

Signed-off-by: Your Name <your.email@provider.com>
```

Mind that the `Signed-off-by: ` tag will be automatically add when you commit the patch using `git commit -s`, and it should be used instead of adjusting things manually.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @jrieck1991!

**Comment by fidencio** [comment]
@chavafg, @egernst, @bergwolf, @devimc, @GabyCT, mind to also take a look at this?

This is changing a default behaviour, but we've introduced a CI to ensure firecracker with jailer works (and the CI will be removed after this is merged), and it does.  And I, maybe naively, fail to see why someone would be running firecracker without the jailer if they can run with the jailer. :-)

**Comment by egernst** [comment]
Maybe a lame question that I should remember -- what was the resolution to the original comment? 

```
# This is disabled by default as additional setup is required
# for this feature today.
```

Ie, what _was_ the extra work, and it is now no longer needed?

**Comment by fidencio** [comment]
> Maybe a lame question that I should remember -- what was the resolution to the original comment?
> 
> ```
> # This is disabled by default as additional setup is required
> # for this feature today.
> ```
> 
> Ie, what _was_ the extra work, and it is now no longer needed?

Would you believe me if I tell you that there was **no** extra work?
When @ananos proposed the patch fixing the `jailer` we cloned the CI, uncommented the `jailer_path` line and it worked as expected (**after** having his patch applied).  So, no CI changes are need, no extra configuration is needed, it simply works.

**Review by jodh-intel — APPROVED** [review]
Thanks @jrieck1991.

lgtm

**Pull request merged**
